src/passes/OptimizeInstructions.cpp

```diff
@@ -1260,16 +1270,26 @@ struct OptimizeInstructions
       return right;
     }
     // x == 0   ==>   eqz x
-    if ((matches(curr, binary(Abstract::Eq, any(&left), ival(0))))) {
-      return builder.makeUnary(EqZInt64, left);
+    if (matches(curr, binary(Abstract::Eq, any(&left), ival(0)))) {
+      return builder.makeUnary(Abstract::getUnary(left->type, Abstract::EqZ),
+                               left);
```

Comment:
It was wrong and conflict with [early rule](https://github.com/WebAssembly/binaryen/blob/master/src/passes/OptimizeInstructions.cpp#L248). Now it generalized

Reply:
Good catch!